### PR TITLE
fix(sre): exit non-zero when library index validation fails

### LINKED ISSUE #871: fix(sre): validate_library_indexes exits 0 on validation failure — CI gate is a no-op
## Summary

`scenarios/sre/scripts/validate_library_indexes.py` silently swallows `ValidationError` and always exits `0`, making the `validate-library` Makefile target (and the `validate-documentation.yaml` CI workflow) a no-op — schema regressions are never caught.

## Affected file

`scenarios/sre/scripts/validate_library_indexes.py` — `load_and_validate_library_index()` (lines 27-30)

```python
try:
    validator.validate(index)
except ValidationError as e:
    logger.exception(f"validation failed: {e}")   # logs, then continues silently
```

`main()` returns normally on every code path, so the process always exits `0` regardless of how many indexes fail validation.

## Impact

- `make validate-library` passes even when every generated index is invalid.
- `validate-documentation.yaml` CI step passes even when schema regressions are introduced.
- The entire generated catalog can contain invalid data without any CI signal.

## Expected behaviour

`validate-library` (and CI) should exit non-zero as soon as any index fails validation, surface all failures, and block the workflow.

## Fix

1. Return a failure count from `load_and_validate_library_index`.
2. Accumulate counts in `main()`.
3. Call `sys.exit(1)` if the total is non-zero.
4. Add a unit test asserting non-zero exit when a `ValidationError` is raised.

## Steps to reproduce

```bash
cd scenarios/sre
# Corrupt any generated index, then:
make validate-library   # exits 0 — bug
echo $?                 # prints 0
```

### PR DESCRIPTION
**Pull request by gianni-di-noia**
fixes #871

### CODE REVIEW

File: scenarios/sre/scripts/validate_library_indexes.py [hunk 1]
```diff
@@ -28,6 +30,9 @@ def load_and_validate_library_index(registry: Registry, index_directory: Path, s
             validator.validate(index)
         except ValidationError as e:
             logger.exception(f"validation failed: {e}")
+            failures += 1
```
Comment (Red-GV) [inline scenarios/sre/scripts/validate_library_indexes.py]: Rather than return the total failures, perhaps let's just call `sys.exit` here with a error message.

**Review by Red-GV — CHANGES_REQUESTED** [review]
Hello, thanks for addressing this! Just a small nit.

File: scenarios/sre/scripts/validate_library_indexes.py (hunk 1 above)
Comment (gianni-di-noia) [reply scenarios/sre/scripts/validate_library_indexes.py]: Thanks! I kept the counter intentionally so all failing indexes are reported in a single run rather than stopping at the first one. Happy to switch if you prefer the simpler form.

Comment (Red-GV) [reply scenarios/sre/scripts/validate_library_indexes.py]: Yes, let's go with the simpler form. Ideally, users are encourage to submit faults/scenarios per PR rather than a bunch because it makes it easier to review. So, in that state, I don't imagine a lot of users will be having multiple validation errors across files.

**Review by Red-GV — APPROVED** [review]
lgtm
